Omr installation docs (#39593)
openshift/openshift-docs: modules/mirror-registry-remote.adoc

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-remote_{context}"]
= Mirroring on a remote host with mirror registry for Red Hat Openshift

> This procedure introduces how to install the _mirror registry for Red Hat Openshift_ on a remote host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.

Reviewer: ```suggestion
This procedure explains how to install the _mirror registry for Red Hat Openshift_ on a remote host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.
```

[NOTE]
====

> Installing the mirror registry for Red Hat Openshift using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.

Reviewer: ```suggestion
Installing the mirror registry for Red Hat Openshift using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an automatically generated password. All access credentials are printed at the end of the install routine.
```

====

.Procedure

> . Download the `mirror-registry.tar.gz` package for the latest version of mirror registry found on the link:https://catalog.redhat.com/software/containers/openshift/mirror-registry-rhel8[mirror registry for Red Hat Openshift] page.

Reviewer: ```suggestion
. Download the `mirror-registry.tar.gz` package for the latest version of mirror registry for Red Hat OpenShift found on the link:https://catalog.redhat.com/software/containers/openshift/mirror-registry-rhel8[mirror registry for Red Hat Openshift] page.
```

+
[source,terminal]
----
$ sudo ./mirror-registry install -v \
  --targetHostname <host_example_com> \
  --targetUsername <example_user> \
  -k ~/.ssh/my_ssh_key \
  --quayHostname <host_example_com> \
  --quayRoot <example_directory_name>
----

. Use the user name and password generated during installation to log into the mirror registry by running the following command:
+
[source,terminal]
----
$ podman login -a pull-secret.txt \
  -u init \
  -p <password> \
  <host_example_com>:8443> \
  --tls-verify=false <1>
----
<1> You can avoid running `--tls-verify=false` by configuring your system to trust the generated rootCA certificates. See "Using SSL to protect connections to Red Hat Quay" and "Configuring the system to trust the certificate authority" for more information.
+
[NOTE]
====
You can also log in by accessing the UI at `\https://<host.example.com>:8443` after installation.
====

. You can mirror {product-title} images after logging in. Depending on your needs, see either the "Mirroring the OpenShift Container Platform image repository" or the "Mirroring Operator catalogs for use with disconnected clusters" sections of this document.
+
[NOTE]
====

> If there are issues with images stored by mirror registry due to storage layer problems, customers are encouraged to either re-mirror the {product-title} images, or re-install mirror registry on more stable storage.

Reviewer: Same comment as above on the other instance of this note

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-remote_{context}"]

> = Remote mirroring with mirror registry for {product-title}

Reviewer 2: Since this is a procedural module, the title should begin with a gerund.

This procedure introduces how to install Red Hat Quay on a remote host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.

[NOTE]
====
Installing Red Hat Quay using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.
====

> 

Reviewer 2: Add a `.Procedure` block title here.

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-remote_{context}"]
= Mirroring on a remote host with mirror registry for {product-title}

Remote mirroring with mirror registry for {product-title}

> This procedure introduces how to install Red Hat Quay on a remote host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.

Reviewer 3: ```suggestion
This procedure introduces how to install the mirroring on a remote host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.
```

Author: This procedure introduces how to install the mirror registry on a remote host using the `mirror-registry` installer tool. By doing so, users can create a local host registry running on port 443 for the purpose of storing a mirror of {product-title} images.

[NOTE]
====

> Installing Red Hat Quay using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.

Reviewer 3: ```suggestion
Installing the mirror registry using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.
```

====

.Procedure

> . Download the `mirror-registry.tar.gz` package for the latest version of mirror registry found on the link:https://console.redhat.com/openshift/downloads[*Downloads*] page of {product-title}.

Reviewer 3: We need to mention online and offline variants of the tarball here.

Author: My understanding is that we’re definitely not building the online installer downstream

Author: Update: my understanding after a conversation with Dave is that there is only a single download pack available. I've made changes accordingly, to address a single download package (which is the offline package that pulls the required images if `image-archive.tar` isn't available).

> . Install Red Hat Quay on a remote host using the `mirror-registry` tool:

Reviewer 3: ```suggestion
. Install the mirror registry on a remote host using the `mirror-registry` tool:
```

+
[source,terminal]
----
$ sudo ./mirror-registry install -v \
  --targetHostname <host_example_com> \
  --targetUsername <example_user> \
  -k ~/.ssh/my_ssh_key \
  --quayHostname <host_example_com> \
  --quayRoot <example_directory_name>
----

> . The Red Hat Quay console is accessible at `\https://<host.example.com>:8443` after installation. Use the user name and password generated during installation to log into the registry by running the following command:

Reviewer 3: ```suggestion
. The mirror registry console is accessible at `\https://<host.example.com>:8443` after installation. Use the user name and password generated during installation to log into the registry by running the following command:
```

Author: Changed to match the local install login procedure.

+
[source,terminal]
----
$ podman login -u init \
  -p <password> \
  --tls-verify=false <1>
----
<1> You can avoid running `--tls-verify=false` by configuring your system to trust the generated rootCA certificates. See "Using SSL to protect connections to Red Hat Quay" and "Configuring the system to trust the certificate authority" for more information.

. You can mirror {product-title} images after logging in. Depending on your needs, see either the "Mirroring the OpenShift Container Platform image repository" or the "Mirroring Operator catalogs for use with disconnected clusters" sections of this document.
+
[NOTE]
====
If there are issues with images stored by mirror registry due to storage layer problems, customers are encouraged to either re-mirror the {product-title} images, or re-install mirror registry on more stable storage.
====

> . You can uninstall Red Hat Quay from a remote host by running the following command:

Reviewer 3: Same comment as in the local install section about separating out uninstall.
